structure/hashmap/hashmap.go

```diff
@@ -13,119 +13,100 @@ type node struct {
 	next  *node
 }
 
-// HashMap is golang implementation of hashmap
+// HashMap is a Golang implementation of a hashmap
 type HashMap struct {
 	capacity uint64
 	size     uint64
 	table    []*node
 }
 
-// New return new HashMap instance
+// New returns a new HashMap instance
 func New() *HashMap {
 	return &HashMap{
 		capacity: defaultCapacity,
 		table:    make([]*node, defaultCapacity),
 	}
 }
 
-// Make creates a new HashMap instance with input size and capacity
-func Make(size, capacity uint64) HashMap {
-	return HashMap{
+// Make creates a new HashMap instance with the specified size and capacity
+func Make(size, capacity uint64) *HashMap {
```

Comment:
Since it returns a pointer now, rename the function to `New`.

Reply:
Renamed function to New on a new commit